src/geometry/adjoints.jl

```diff
@@ -0,0 +1,52 @@
+using Base: Tuple
+function ChainRulesCore.rrule(::typeof(Interpolations.BSplineInterpolation), T, A, it, axs)
+    y = Interpolations.BSplineInterpolation(T, A, it, axs)
+    return (y, Δ -> begin
+        (NoTangent(), NoTangent(), Δ, NoTangent(), NoTangent())
+    end)
+end
+
+function ChainRulesCore.rrule(::typeof(Interpolations.BSplineInterpolation{T, N, TA, IT, TAX}), A, axs, it) where {T, N, TA, IT, TAX}
+    y = Interpolations.BSplineInterpolation{T, N, TA, IT, TAX}(A, axs, it)
+    return (y, Δ -> begin
+        (NoTangent(), Δ, NoTangent(), NoTangent())
+    end)
+end
+
+function ChainRulesCore.rrule(::typeof(Interpolations.Extrapolation{T, N, ITPT, IT, ET}), itp, et::Union{Interpolations.Flat, Interpolations.Reflect, Interpolations.Line}) where {T,N,ITPT,IT,ET}
+    y = Interpolations.Extrapolation{T,N,ITPT,IT,ET}(itp, et)
+    return (y, Δ -> begin
+        (NoTangent(), Δ, NoTangent())
+    end)
+end
+
+Zygote.@nograd Interpolations.tweight
+Zygote.@nograd Interpolations.tcoef
```

Comment:
Also transform this into ChainRules' language because ChainRules doesn't recognize Zygote.

Reply:
I think the `@non_diff` notation is slightly verbose and Zygote preferentially chooses its own rules which can be better optimised in cases, so it's alright, but you should add the CRC method in here as well.